src/domain/resource/utils.js

```diff
@@ -376,6 +404,27 @@ export const isDateReservable = (resource, date) => {
   return isAdmin || (isBefore && isAfter);
 };
 
+const isBetweenReservedTimeRange = (events, start, end) => {
+  const overlapped = events.find((event) => {
```

Comment:
`Array.prototype.some` could be used instead of find, it works as find but returns a boolean.